### ISSUE TITLE
Help needed for www.pombase.org/pombemine configuration

### ISSUE THREAD

**Issue by kimrutherford** [intermine/pombemine#66]
Hi.

I've added a ProxyPass rule for pombemine on www.pombase.org:

```ApacheConf
<Location /pombemine>
  ProxyPass http://pombemine.rahtiapp.fi/pombemine retry=10
  ProxyPassReverse http://pombemine.rahtiapp.fi/pombemine
</Location>
```

The initial page is retrieved correctly by the browser but the other URLs don't have the correct prefix:

So for example this:
```
https://www.pombase.org/js/compiled/app-fc93e4c.js
```
should be:
```
https://www.pombase.org/pombemine/js/compiled/app-fc93e4c.js
```

Is it possible to configure the PombeMine front end to prefix URLs with `https://www.pombase.org/pombemine/` rather than `https://www.pombase.org/`?

If you'd like to test anything it's here: https://www.pombase.org/pombemine/

Thanks!

**Comment by danielabutano** [comment]
Hi @kimrutherford ,
I have tried locally using nginx listening at port 80 (with biotestmine available at http://localhost:5000/biotestmine)
server {
  listen 80       default_server;
  server_name     localhost;
  root /var/www/html;

  location /biotestmine {
    proxy_pass        http://localhost:5000/biotestmine;
    proxy_set_header  X-Real-IP $remote_addr;
    proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header  Host $http_host;
  }

  location / {
    proxy_pass        http://localhost:5000;
    proxy_set_header  X-Real-IP $remote_addr;
    proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header  Host $http_host;
  }

So I think you need also to configure the location /

**Comment by kimrutherford** [comment]
> So I think you need also to configure the location /

Hi Daniela.  I think that will interfere with the main pombase.org website because it will proxy everything to PombeMine.  I'd like to just proxy the paths under the /pombemine because everything else is proxied to the main PomBase webapp.

I noticed that AllianceMine uses a prefix and all the resources are requested using a /alliancemine prefix.  Can we do something similar for PombeMine?

**Comment by danielabutano** [comment]
yeah, you are right.
@heralden do you have any suggestion?

**Comment by heralden** [comment]
Yes, the `BLUEGENES_DEPLOY_PATH` config var is for this (https://github.com/intermine/bluegenes/blob/dev/docs/configuring.md). Note that if you set it to `/pombemine`, you will get the path `/pombemine/pombemine`. You might want to use something else like /intermine /im etc.

**Comment by kimrutherford** [comment]
> Note that if you set it to /pombemine, you will get the path /pombemine/pombemine. You might want to use something else like /intermine /im etc.

Hi.  Could you explain this more?  Which path will be "/pombemine/pombemine"?

**Comment by danielabutano** [comment]
I can do the configuration, but I need to redeploy ...non sure if i want to do before tomorrow presentaation.....

**Comment by heralden** [comment]
> Hi. Could you explain this more? Which path will be "/pombemine/pombemine"?

Everything after www.pombase.org

**Comment by ValWood** [comment]
I agree not to upadate before the demo.  It  can be done for the announcement to the community.

pombemine/pombemine looks a bit weird, but I guess if that is the only option.......

**Comment by kimrutherford** [comment]
Is there a redirect from www.pombase.org/pombemine to www.pombase.org/pombemine/pombemine within the app?

**Comment by heralden** [comment]
Yes, that's right. Just to clarify, `/pombemine` would be the deploy path while the second `/pombemine` would be the name of the mine. This is because BG supports connecting to other mines.

**Comment by kimrutherford** [comment]
OK, thanks.  Your suggestion of /im looks quite tidy.

**Comment by danielabutano** [comment]
Hi @heralden, i have changed the pombemine-production.yaml, adding 
`deployPath: im`
But we need to change also the template deployment.yaml to add:
- name: BLUEGENES_DEPLOY_PATH
              value: {{ .Values.bluegenes.deployPath }}

Question: can I do locally and then run the upgrade, or I need to upload something?
Thanks!

**Comment by heralden** [comment]
Yes, you're right. You can edit the template locally then run the upgrade. Once you confirmed it's working, you can commit it. Just make sure to set `deployPath: /im` (starting with a slash).
